lib/fake_data.rb

```diff
@@ -0,0 +1,115 @@
+require "csv"
+
+class FakeData
+  def self.insert
+    new.insert
+  end
+
+  def self.delete
+    new.delete
+  end
+
+  def insert
+    if has_test_subscribers?
+      puts "There is already test data in the system."
+      puts "Run rake fake_data:delete first."
+      return
+    end
+
+    fake_subscriptions_data.each do |subscription_stat|
+      subscriber_ids = create_subscribers(subscription_stat.no_subscribers).ids
+      create_subscriptions(subscriber_ids, subscription_stat.no_subscriptions)
+    end
+
+    count = existing_test_subscribers.count
+    puts "Created #{count} subscribers."
+  end
+
+  def delete
+    count = existing_test_subscribers.delete_all
+    puts "Deleted #{count} subscribers."
+  end
+
+private
+
+  def frequencies
+    @frequencies ||= begin
+      Enumerator.new do |yielder|
+        loop do
+          3.times { yielder << :weekly }
+          7.times { yielder << :daily }
+          10.times { yielder << :immediately }
+        end
+      end
+    end
+  end
+
+  def subscriber_list_ids
+    @subscriber_list_ids ||= begin
+      Enumerator.new do |yielder|
+        loop do
+          SubscriberList.order("RANDOM()").pluck(:id).each do |id|
+            yielder << id
+          end
+        end
+      end
+    end
+  end
+
+  def uuids
+    Enumerator.new do |yielder|
+      loop { yielder << SecureRandom.uuid }
+    end
+  end
+
+  def create_subscribers(count)
+    puts "Building #{count} subscribers..."
+
+    columns = %i(address)
+    records = uuids.take(count).map do |uuid|
+      ["success+#{uuid}@simulator.amazonses.com"]
+    end
+
+    puts "> Importing #{records.count} subscribers..."
+
+    Subscriber.import!(columns, records, validate: false)
+  end
+
+  def create_subscriptions(subscriber_ids, count)
+    count = subscriber_ids.count * count
+
+    puts "> Building #{count} subscriptions..."
+
+    columns = %i(subscriber_id subscriber_list_id frequency uuid)
+
+    count.times do
+      records = subscriber_ids.zip(subscriber_list_ids, frequencies, uuids)
+
+      puts ">> Importing #{records.count} subscriptions..."
+
+      Subscription.import!(columns, records, validate: false)
+    end
+  end
+
+  def existing_test_subscribers
+    Subscriber.where("address LIKE 'success+%@simulator.amazonses.com'")
+  end
+
+  def has_test_subscribers?
+    existing_test_subscribers.exists?
+  end
+
+  SubscriptionStat = Struct.new(:no_subscriptions, :no_subscribers)
```

Comment:
I think `number_of_` would be clearer here.

Reply:
Fixed in 356b7a9